ament_cmake_cppcheck/cmake/ament_cmake_cppcheck_lint_hook.cmake

```diff
@@ -33,18 +33,36 @@ if(_source_files)
   if(NOT CMAKE_VERSION VERSION_LESS "3.7.0")
     get_directory_property(_build_targets DIRECTORY ${CMAKE_SOURCE_DIR} BUILDSYSTEM_TARGETS)
     foreach(_target ${_build_targets})
-      get_property(_include_dirs
+      # Check if INCLUDE_DIRECTORIES is defined first in case the target is an INTERFACE
+      get_property(_include_dirs_defined
         TARGET ${_target}
         PROPERTY INCLUDE_DIRECTORIES
+        DEFINED
       )
 
+      if(_include_dirs_defined)
+        get_property(_include_dirs
+          TARGET ${_target}
+          PROPERTY INCLUDE_DIRECTORIES
+        )
+      else()
+        # Target might be an interface
+        get_property(_include_dirs
+          TARGET ${_target}
+          PROPERTY INTERFACE_INCLUDE_DIRECTORIES
+        )
+      endif()
+
       # Only append include directories that are from the package being tested
       # This accomplishes two things:
       #     1. Reduces execution time (less include directories to search)
       #     2. cppcheck will not check for errors in external packages
       foreach(_include_dir ${_include_dirs})
-        string(REGEX MATCH "^${CMAKE_SOURCE_DIR}.*" _is_match ${_include_dir})
-        if(_is_match)
+        # Check if include directory is a subdirectory of the source directory
+        string(REGEX MATCH "^${CMAKE_SOURCE_DIR}.*" _is_subdirectory ${_include_dir})
```

Comment:
This regex will also match sibling directories of e.g packages which start with the same name as this directory. So the regex should have a trailing shash before the wildcard.

Also this might need to consider platform specific path separators.

Same below.

Reply:
This seems to be a way to do it 0472892
Open to suggestions.

Reply:
My impression was that CMake prefers paths to be stored with forward slashes internally and then converts them on platforms that need it but I can't find the relevant documentation.

Reply:
Confirmed on a Windows machine that include dirs are cmake paths. I’ll remove the native path command.